src/index.ts

```diff
@@ -24,13 +24,96 @@ export function run(
   files: malloy.URLReader,
   args: string[]
 ): Promise<malloy.Result> {
-  const connection = new BigQueryConnection("bigquery");
+  const connection = new DuckDBConnection("duckdb");
   const runtime = new malloy.Runtime(files, connection);
+
   const { query, model } = getOptions(args);
+
+  // Current way to load a model and a query:
   const queryMaterializer = model
     ? runtime.loadModel(model).loadQuery(query)
     : runtime.loadQuery(query);
+
   return queryMaterializer.run();
+
+  // Rename "runtime" to "malloy"?
+  // Why do we need ModelMaterializer and QueryMaterializer? Can it just be Model and Query in the API?
+
+  // Things I want to do:
```

Comment:
other things we might need (especially if you're trying to build an app out of this API):

- get dialect
- get SQL (instead of run and get results)
- get row limit
- serialize/deserialize Malloy objects
- get expected structure of result data (but don't run query)
- get list of items in a source (joins, fields, etc), get properties of those items
- (there's probably more)

Reply:
Great list. I'll move this into a separate doc so it's a little more discoverable and easier to share/discuss